task-4.ts

```diff
@@ -0,0 +1,43 @@
+class Product {
+  constructor(public name: string, public price: number, public stock: number) {
+    this.name = name;
+    this.price = price;
+    this.stock = 0;
+  }
+
+  addStock(quantity: number) {
+    this.stock += quantity;
+  }
+
+  reduceStock(quantity: number) {
+    if (this.stock - quantity > 0) {
+      this.stock -= quantity;
+    }
+  }
+}
+
+class Customer {
+  constructor(
+    public name: string,
+    public email: string,
+    public cart: Product[]
+  ) {
+    this.cart = cart;
+    this.email = email;
+    this.name = name;
+  }
+
+  addToCart(product: Product) {
+    if (!product) return false;
```

Comment:
you should check whether stock of product is more than zero. If there is product in stock product.getStocks() == 0 then you shouldn't add it to cart